src/Compilers/CSharp/Portable/BoundTree/DecisionTreeBuilder.cs

```diff
@@ -78,16 +83,13 @@ protected DecisionTree AddToDecisionTree(DecisionTree decisionTree, SyntaxNode s
             }
         }
 
-        protected delegate DecisionTree DecisionMaker(
+        protected delegate DecisionTree.Guarded DecisionMaker(
```

Comment:
>protected delegate DecisionTree.Guarded DecisionMaker( [](start = 8, length = 54)

Consider adding a comment about the purpose of this delegate, when it it invoked and what is it supposed to do.